packages/synapse-sdk/src/telemetry/utils.ts

```diff
@@ -0,0 +1,61 @@
+import type * as SentryBrowser from '@sentry/browser'
+import type * as SentryNode from '@sentry/node'
+import { createError as originalCreateError } from '../utils/errors.ts'
+import { getGlobalTelemetry } from './singleton.ts'
+
+// Dynamically import the correct Sentry package based on environment
+export const isBrowser =
+  typeof (globalThis as any).window !== 'undefined' && typeof (globalThis as any).document !== 'undefined'
+
+export type Sentry = typeof SentryNode.default | typeof SentryBrowser.default
+
+export async function getSentry(): Promise<{ Sentry: Sentry; integrations: any[] }> {
```

Comment:
```suggestion
/**
 * Dynamically import the correct Sentry package for whether we're running in a browser or Node.
 */
export async function getSentry(): Promise<{ Sentry: Sentry}> {
```

(I'll leave a comment about this in service.ts but I think it makes more sent to have the integration definitions happen in service.ts rather than being tucked away here.)